tests/overrides/test_data_parallel.py

```diff
@@ -0,0 +1,55 @@
+from unittest.mock import MagicMock
+
+import pytest
+import torch
+
+from pytorch_lightning.overrides.data_parallel import LightningDistributedModule
+
+
+def test_lightning_distributed_module_methods():
+    """ Test that the LightningDistributedModule redirects .forward() to the LightningModule methods. """
+    pl_module = MagicMock()
+    dist_module = LightningDistributedModule(pl_module)
+
+    batch = torch.rand(5)
+    batch_idx = 3
+
+    pl_module.training = True
+    pl_module.testing = False
```

Comment:
curious how the module has these attributes, isn't it Trainer?

Reply:
see my answer [here](https://github.com/PyTorchLightning/pytorch-lightning/pull/5185#discussion_r555001405)
The property is copied to the model [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/404af43cde6696d04bb1899da2bb7e334e49716d/pytorch_lightning/trainer/connectors/model_connector.py)
happy to follow up on this in a next step. The test that I wrote there is just to make sure it works the same as before refactor.